File: Authorizer/SMSAuthorizer.py
from .Authorizer import Authorizer

class SMSAuthorizer(Authorizer):
  authorized = False
  
  def verify_code(self, code):
    print(f"Verifying 2FA code {code}")
    self.authorized = True

  def is_authorized(self) -> bool:
    return self.authorized == True

File: src/Authorizer/SMSAuthorizer.ts
import Authorizer from "./Authorizer"

export default class SMSAuthorizer implements Authorizer {
    private authorized = false

    public check_authorization(security_code: number): void {
        console.log(`Checking authorization for code ${security_code}`)
        this.authorized = true
    }

    public is_authorized() {
        return this.authorized === true
    }
}
